### DISCUSSION TITLE
Create mesh from PyTorch Tensor

### DISCUSSION THREAD

**Discussion by supertan0204** [mitsuba-renderer/mitsuba3#1223] [Q&A]
Hi, I'm trying to load a mesh in Mitsuba 3 using data from PyTorch. Specifically, I have a tensor `v` with shape `(num_vertices, 3)` which contains vertex position data, and a tensor `f` with shape `num_faces, 3` which contains face data. How should I construct a `mi.Mesh` object with these data?

**Comment by njroussel** [comment]
Hi @supertan0204 

I'd recommend you have a look at this guide: https://mitsuba.readthedocs.io/en/stable/src/how_to_guides/mesh_io_and_manipulation.html#Adding-and-editing-attributes

We show to procedurally generate a `mi.Mesh`, which requires writing known vertex and face data to separate buffers.
